monix-reactive/shared/src/test/scala/monix/reactive/internal/operators/RecursiveConcatSuite.scala

```diff
@@ -50,4 +50,15 @@ object RecursiveConcatSuite extends BaseOperatorSuite {
     val f = range(0, count).sumL.runAsync; s.tick()
     assertEquals(f.value, Some(Success(count.toLong * (count - 1) / 2)))
   }
+
+  val nats: Observable[Long] = Observable.defer {
```

Comment:
Ideally we could just do `Observable.now(1L) ++ nats.map(_ + 1)` - if `Observable.defer` is needed then maybe we could just include it in the implementation? So we could even leave old `ConcatObservable` but pass suspended `Observable`

Reply:
Yeah, I was following the convention from `def range` above - but I think with `++` having a lazy rhs it doesn't make sense to defer either of these observables. Unless we make `range` use `concat` instead too...

Reply:
I mean that we could leave `ConcatObservable` strict (and use it in `concat` implementation if possible - #644) implementing `++` like:
```scala
new ConcatObservable(self, Observable.suspend(other))
```
What do you think?

Reply:
it could work - I'll see

Reply:
It works, but I'm having some random SOs from time to time - looking for a reason

Reply:
Apparently it has something to do with `defer` not trampolining the function passed to it - but I'm not sure about the details